packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1457,16 +1433,8 @@ export class GarbageCollector implements IGarbageCollector {
                 });
             }
 
-            // If SweepReady Usage Detection is enabed, close the main container
-            // Once Sweep is fully implemented, this will be removed since the objects will be gone
-            // and errors will arise elsewhere in the runtime
-            if (state === UnreferencedState.SweepReady &&
-                sweepReadyUsageDetectionSetting.read(this.mc.config).mainContainer
-            ) {
-                const error = new SweepReadyUsageError(
-                    "SweepReady object used in Non-Summarizer Client",
-                    { errorDetails: JSON.stringify(propsToLog) });
-                this.runtime.closeFn(error);
+            if (state === UnreferencedState.SweepReady) {
+                this.sweepReadyUsageHandler.usageDetectedInMainContainer(propsToLog);
```

Comment:
You should add `usageType` here so we know whether the object was loaded, revived or used.

Reply:
Good catch